client/dashboard/sites/use-app-context-for-sites-list.tsx

```diff
@@ -0,0 +1,81 @@
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import { useAppContext } from '../app/context';
+import AddNewSite from './add-new-site';
+
+interface AppContextForSitesListReturn {
```

Comment:
All of these customisations, except `pageTitle`, are related to the button. The modal appears when the button is clicked. The modal's open state is only relevant when there is a button. Perhaps this should just give the ability to pass a whole custom button component.

So it could either be two new props added to the `AppConfig` interface:
- `newSiteButton: ReactNode`
- `siteListPageTitle: string`

Or it could be done using filters and a slot fill.